pkg/controller/podautoscaler/replica_calculator.go

```diff
@@ -245,3 +337,20 @@ func (c *ReplicaCalculator) GetMetricReplicas(currentReplicas int32, targetUtili
 	// however many replicas factored into our calculation
 	return int32(math.Ceil(newUsageRatio * float64(len(metrics)))), utilization, timestamp, nil
 }
+
+// GetObjectMetricReplicas calculates the desired replica count based on a target metric utilization (as a milli-value)
+// for the given object in the given namespace, and the current replica count.
+func (c *ReplicaCalculator) GetObjectMetricReplicas(currentReplicas int32, targetUtilization int64, metricName string, namespace string, objectRef *autoscaling.CrossVersionObjectReference) (replicaCount int32, utilization int64, timestamp time.Time, err error) {
```

Comment:
there isn't one because the existing replica calc test is structured around the actual metrics client, so it needs a working implementation of `GetObjectMetrics`, which we don't have (it just returns a "not-implemented" error).  I can either a) restructure the entire test, or b) add in the test in part 3 of this PR, which adds support for custom metrics using the custom metrics API types.